### PR TITLE
Setup Census Block Group download and GeoJSON encoding

### PR DESCRIPTION
**Pull request by esfoobar-usds**
This PR also sets up Jupyter Notebooks. See the README file in the `/score` directory.

### CODE REVIEW

File: score/scripts/download_cbg.py
Comment (NatHillardUSDS) [inline score/scripts/download_cbg.py:10]: you can use `pathlib` to work with paths in an OS-, and root-directory independent way - more [here](https://realpython.com/python-pathlib/) but its a lifesaver

Comment (NatHillardUSDS) [reply score/scripts/download_cbg.py:10]: Additionally you can use importlib.resources (more [here](https://realpython.com/python-import/#introducing-importlibresources)) to treat the filesystem like a package, allowing you to run the script from any location (including later direct from within a docker container)

**Review by NatHillardUSDS — CHANGES_REQUESTED** [review]
Overall awesome code, awesome setup, awesome documentation, and excellent first pass!! Very clear and very pared down, this is really great and very methodical.

I know this is a WIP; I ran through the docs to make sure I could get things going locally and was able to do so successfully -- no problems here, ran on the first try. This is a testament to good docs!!

I commented a few things, which could be seen as incremental improvements (though `multiprocessing` does improve our quality of life a fair amount).

There was one true bug I found: in the os.remove check for file existence.

Other than this, I would recommend the following, which given that this is in progress you may already have planned, but wanted to call out in case. None of these are critical / blocking merge but in the case of the `p1`s would improve performance a fair amount:

p1:
* `multiprocessing` to parallelize downloads and/or processing 
* dockerizing this script itself and installing necessary tooling dependencies rather than running the tool from a given container on each run
* working directly with zipfiles to speed up processing

p2
* factoring into functions and/or class (makes multiprocessing easier also)
* using `pathlib` and `importlib` for fs- and OS-independent path processing 
* using the python `gdal` library directly for better integration with arguments

p3:
* use Poetry for dependency and package management 

anyhow, overall thank you for documenting all of this and it's great you were able to get this going in such short notice!

File: score/README.md
Comment (esfoobar-usds) [reply score/README.md:11]: Yep, we can evaluate later.

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:7]: I also thing this could be revisited later -- we/the user won't be downloading census data often. This script would be typically be run the very first time you setup the repo locally.

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:30]: Ah yes, I saw this on Mike's notes. Again, since this is run typically once, we can improve later.

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:47]: Agreed. But until we don't know better the organization when we jump to lambdas, we can re-evaluate then.

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:58]: Yeah, since this is running on a local computer, I don't think we will pass too many params.

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:62]: Makes sense. Again let's see what it looks like on Lambdas and leave as is for now.

File: score/scripts/generate_mbtiles.py
Comment (esfoobar-usds) [reply score/scripts/generate_mbtiles.py:14]: Saw the same thing, yeah. I'll try that flag.

File: score/scripts/generate_mbtiles.py
Comment (esfoobar-usds) [reply score/scripts/generate_mbtiles.py:4]: Fixed in new commit

File: score/scripts/generate_mbtiles.py
Comment (esfoobar-usds) [reply score/scripts/generate_mbtiles.py:14]: Hmm -- seems we need to pass a projection type after the `-s` flag -- any idea what that would be? It does seems to be a warning, so I'll resolve for now and we can address this later.

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:10]: Addressed in latest commit of PR

**Comment by esfoobar-usds** [comment]
@NatHillardUSDS can you review and approve if it looks okay (I addressed your comments but let me know if you disagree or want to do more changes)

File: score/scripts/download_cbg.py
Comment (NatHillardUSDS) [inline score/scripts/download_cbg.py:8]: Even cooler pathlib does operator overloading with "/" so you can do 
```python
Path.cwd() / 'data'
```
optional though

File: score/scripts/generate_mbtiles.py
Comment (NatHillardUSDS) [inline score/scripts/generate_mbtiles.py:15]: pathlib gives a `.rmdir()` so you wouldn't need `shutil`

**Review by NatHillardUSDS — APPROVED** [review]
I'd overall say all the `joinpath`s can be converted to the short-hand, slash based syntax to make things a bit neater, and `shutil` can also be swapped out for `pathlib`, though this is optional.

For the followup items, agreed these aren't critical for now, but is there a ticket to capture tackling them later? Want to make sure we log planned improvements if we don't have them already. 

Otherwise approved though!

File: score/scripts/generate_mbtiles.py
Comment (esfoobar-usds) [reply score/scripts/generate_mbtiles.py:15]: Oh, I thought I tried it but it complained the directory wasn't empty? Maybe in Windows?

File: score/scripts/download_cbg.py
Comment (esfoobar-usds) [reply score/scripts/download_cbg.py:8]: Oh, that would be much cleaner, let me do it quick.

File: score/scripts/generate_mbtiles.py
Comment (NatHillardUSDS) [reply score/scripts/generate_mbtiles.py]: geoJSON has a default of `EPSG:4326`  ([ref](https://macwright.com/2015/03/23/geojson-second-bite.html)) but individual geojson files can specify an override; the ones generated by this process are specifying `"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:EPSG::4269" } },` - this appears to be the [census default](https://gis.stackexchange.com/questions/27702/what-is-the-srid-of-census-gov-shapefiles). 

So I believe you actually need to specify `-s_srs EPSG:4269 -t_srs EPSG:4326` in the `ogr2ogr` command that generates the geojson from the shapefile as used [here](https://engelsjk.com/posts/zcta/)

File: score/scripts/generate_mbtiles.py
Comment (esfoobar-usds) [reply score/scripts/generate_mbtiles.py:15]: Yup, this fails:

```
-> mvt_tiles_path.rmdir()
(Pdb) n
OSError: [WinError 145] The directory is not empty: 'C:\\opt\\justice40-tool\\score\\data\\tiles\\mvt'
> c:\opt\justice40-tool\score\scripts\generate_mbtiles.py(15)<module>()
-> mvt_tiles_path.rmdir()
```

**Comment by esfoobar-usds** [comment]
@NatHillardUSDS I've created [a ticket](https://github.com/usds/justice40-tool/issues/134) to track pending work.

**Pull request merged**
